src/jit/codegencommon.cpp

```diff
@@ -2698,7 +2708,14 @@ void CodeGen::genExitCode(BasicBlock* block)
 
 void CodeGen::genJumpToThrowHlpBlk(emitJumpKind jumpKind, SpecialCodeKind codeKind, GenTreePtr failBlk)
 {
-    if (!compiler->opts.compDbgCode)
+    bool useThrowHlpBlk = !compiler->opts.compDbgCode;
+
+#if defined(UNIX_X86_ABI) && FEATURE_EH_FUNCLETS
+    // Inline exception-throwing code in funclet to make it possible to unwind funclet frames.
+    useThrowHlpBlk = useThrowHlpBlk && !siInFuncletRegion;
+#endif // UNIX_X86_ABI && FEATURE_EH_FUNCLETS
```

Comment:
Instead of using `!siInFuncletRegion`, which uses a variable specific to the scope info code, you should use `(compiler->funCurrentFunc()->funKind != FUNC_ROOT)`, which is more generally used.